src/natcap/invest/urban_nature_access.py

```diff
@@ -547,6 +655,376 @@ def execute(args):
+            if ok_for_geometries_to_touch and polygon.touches(
+                    shapely_polygon_lookup[intersect_poly_fid]):
+                continue
+
+            if intersect_poly_fid == poly_fid or polygon.intersects(
+                    shapely_polygon_lookup[intersect_poly_fid]):
+                poly_intersect_lookup[poly_fid].add(intersect_poly_fid)
+        polygon = None
+    LOGGER.info(
+        'poly intersection feature lookup 100.0%% complete on %s',
+        os.path.basename(vector_path))
+
+    # Build maximal subsets
+    subset_list = []
+    while len(poly_intersect_lookup) > 0:
+        # sort polygons by increasing number of intersections
+        intersections_list = [
+            (len(poly_intersect_set), poly_fid, poly_intersect_set)
+            for poly_fid, poly_intersect_set in
+            poly_intersect_lookup.items()]
+        intersections_list.sort()
+
+        # build maximal subset
+        maximal_set = set()
+        for _, poly_fid, poly_intersect_set in intersections_list:
+            last_time = pygeoprocessing.geoprocessing._invoke_timed_callback(
+                last_time, lambda: LOGGER.info(
+                    "maximal subset build approximately %.1f%% complete "
+                    "on %s", 100.0 * float(
+                        feature_count - len(poly_intersect_lookup)) /
+                    feature_count, os.path.basename(vector_path)),
+                pygeoprocessing.geoprocessing._LOGGING_PERIOD)
+            if not poly_intersect_set.intersection(maximal_set):
+                # no intersection, add poly_fid to the maximal set and remove
+                # the polygon from the lookup
+                maximal_set.add(poly_fid)
+                del poly_intersect_lookup[poly_fid]
+        # remove all the polygons from intersections once they're computed
+        for poly_fid, poly_intersect_set in poly_intersect_lookup.items():
+            poly_intersect_lookup[poly_fid] = (
+                poly_intersect_set.difference(maximal_set))
+        subset_list.append(maximal_set)
+    LOGGER.info(
+        'maximal subset build 100.0%% complete on %s',
+        os.path.basename(vector_path))
+    return subset_list
+
+
+def _reproject_and_identify(base_vector_path, target_projection_wkt,
+                            target_path, driver_name, id_fieldname):
+    """Reproject a vector and add an ID field.
+
+    Args:
+        base_vector_path (string): The string path to the source vector.
+        target_projection_wkt (string): The WKT of the target projection.
+        target_path (string): The string path to where the new vector should be
+            saved.
+        driver_name (string): The GDAL driver name of the target vector.
+        id_fieldname (string): The name of the ID field.  A new field with this
+            name and an integer type will be created in the target vector.
+            Each feature in the target vector will be assigned a unique integer
+            ID.
```

Comment:
Awesome, I like this workflow in general when handling vectors that don't have a specified unique ID.

Reply:
Yes!  It's always nice to be able to uniquely identify the AOIs, and in this case it turns out to be necessary for when we rasterize them a little later on in `execute`.